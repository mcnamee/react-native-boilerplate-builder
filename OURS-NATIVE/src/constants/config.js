const isDevEnv = process.env.NODE_ENV === 'development';

export default {
  // App Details
  appName: 'AwesomeProject',

  // Build Configuration - eg. Debug or Release?
  isDevEnv,

  // Date Format
  dateFormat: 'Do MMM YYYY',

  // API
  apiBaseUrl: isDevEnv
    ? 'https://www.digitalsupply.co/wp-json/wp'
    : 'https://www.digitalsupply.co/wp-json/wp',


  // Google Analytics - uses a 'dev' account while we're testing
  gaTrackingId: isDevEnv ? 'UA-84284256-2' : 'UA-84284256-1',
};
