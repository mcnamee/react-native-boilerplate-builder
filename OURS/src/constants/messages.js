export const generalMessages = {};

export const successMessages = {
  // Defaults
  defaultForm: 'Success - Form Saved',

  // Member
  login: 'You are now logged in',
  signUp: 'You are now signed up. Please login to continue.',
  forgotPassword: 'Password reset. Please check your email.',
};

export const errorMessages = {
  // Defaults
  default: 'Hmm, an unknown error occured',
  timeout: 'Server Timed Out. Check your internet connection',
  invalidJson: 'Response returned is not valid JSON',
  missingData: 'Missing data',

  // Member
  memberNotAuthd: 'You need to be logged in, to update your profile',
  memberExists: 'Member already exists',
  missingFirstName: 'First name is missing',
  missingLastName: 'Last name is missing',
  missingEmail: 'Email is missing',
  missingPassword: 'Password is missing',
  passwordsDontMatch: 'Passwords do not match',

  // Articles
  articlesEmpty: 'No articles found',
  articles404: 'This article could not be found',
};
