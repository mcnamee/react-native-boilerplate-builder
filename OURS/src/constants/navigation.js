import Colors from '../../native-base-theme/variables/commonColor';

export default {
  navbarProps: {
    navigationBarStyle: { backgroundColor: 'white' },
    titleStyle: {
      color: Colors.textColor,
      alignSelf: 'center',
      fontSize: Colors.fontSizeBase,
    },
    backButtonTintColor: Colors.textColor,
  },

  tabProps: {
    swipeEnabled: false,
    activeBackgroundColor: 'rgba(255,255,255,0.1)',
    inactiveBackgroundColor: Colors.brandPrimary,
    tabBarStyle: { backgroundColor: Colors.brandPrimary },
  },

  icons: {
    style: { color: 'white', height: 30, width: 30 },
  },
};
