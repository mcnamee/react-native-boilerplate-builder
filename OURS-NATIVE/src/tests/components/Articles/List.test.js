import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import ArticlesList from '../../../components/Articles/List';
import { errorMessages } from '../../../constants/messages';

it('<ArticlesList /> shows a nice error message', () => {
  const Component = <ArticlesList list={[]} />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText(errorMessages.articlesEmpty));
});

it('<ArticlesList /> shows a list of articles correctly', () => {
  const listItem = {
    id: 0,
    name: 'ABC',
    excerpt: 'DEF',
    contentRaw: 'DEF',
    date: '22/33/44',
  };

  const Component = <ArticlesList listFlat={[listItem]} />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText(listItem.name));
});
