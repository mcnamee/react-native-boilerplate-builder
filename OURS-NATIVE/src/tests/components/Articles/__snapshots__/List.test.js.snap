// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticlesList /> shows a list of articles correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "padding": 10,
      },
    ]
  }
>
  <RCTScrollView
    ListFooterComponent={null}
    data={
      Array [
        Object {
          "contentRaw": "DEF",
          "date": "22/33/44",
          "excerpt": "DEF",
          "id": 0,
          "name": "ABC",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <TouchableOpacity
            activeOpacity={0.8}
            onPress={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              cardBody={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": -5,
                  "paddingLeft": -5,
                  "paddingRight": -5,
                  "paddingTop": -5,
                }
              }
            />
            <View
              cardBody={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": -5,
                  "paddingLeft": -5,
                  "paddingRight": -5,
                  "paddingTop": -5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-start",
                      "alignSelf": "stretch",
                      "flex": 1,
                    },
                    Object {
                      "paddingHorizontal": 15,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "flex": 1,
                        "height": 10,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "fontWeight": "800",
                      },
                    ]
                  }
                >
                  ABC
                </Text>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "flex": 1,
                        "height": 15,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    }
                  }
                >
                  DEF
                </Text>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "flex": 1,
                        "height": 5,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "height": 20,
        },
      ]
    }
  />
</View>
`;

exports[`<ArticlesList /> shows a nice error message 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 20,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 21,
            "lineHeight": 25,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      Uh oh
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "marginBottom": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      No articles found
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 20,
          },
        ]
      }
    />
  </View>
</View>
`;
